cranelift-wasm/src/code_translator.rs

```diff
@@ -1932,3 +1941,19 @@ pub fn bitcast_arguments(
         }
     }
 }
+
+/// A helper to extract all the `Type` listings of each variable in `params`
+/// for only parameters the return true for `is_wasm`, typically paired with
+/// `is_wasm_return` or `is_wasm_parameter`.
+pub fn wasm_param_types(
+    params: &[ir::AbiParam],
+    mut is_wasm: impl FnMut(usize) -> bool,
```

Comment:
Can you comment on why this is `mut` and `Mut`? The ultimate function should be a non-mutating predicate.

Reply:
Oh this is mostly just a Rust idioms thing where you idiomatically take `FnMut` since it's the most flexible. For brevity and simplicity though I've switched this to `Fn`, since `FnMut` wasn't actually needed.